### ISSUE TITLE
GitHub mcp file creation error: Failed to call tool create_or_update_file: Error: MCP error 32604:

### ISSUE THREAD

**Issue by Mahshadn** [modelcontextprotocol/servers#428]
I get this error when Claude Desktop is trying to create files to my repo. I created this private repo in the same chat earlier.
Claude can access the repo, however, when asking to create files (json, yaml, etc.) it doesn't work and returns below error:

Failed to call tool create_or_update_file: Error: MCP error -32604: Invalid arguments: content.encoding: Required, content.content: Required

**Comment by txbm** [comment]
I get this quite frequently as well. The only thing that works currently is when it uses push_files

**Comment by Rehcywgnaf** [comment]
did you try to use create_or_update_file instead of push_files ? maybe a turnaround ? 
I get the same 32603 or 32604 error but for now waiting for my claude to comeback

**Comment by txbm** [comment]
Oh forgot to update here but I think I fixed this and many other problems w/ Github MCP server in this PR https://github.com/modelcontextprotocol/servers/pull/434

**Comment by Rehcywgnaf** [comment]
I get this error : 
MCP error -32603: Invalid arguments: content.encoding: Required, content.content: Required
I dont know if this PR can help me, does it ?

**Comment by KTreePortfolio** [comment]
I have the same error while I am able to push files.   Using windows and Claude Desktop.
I suspect an error in the permissions github side or  problems with the new type of token.
I am testing the old one now. With the new type it failed to call create_or_update  due to -32603 content.content error

**Comment by txbm** [comment]
Yes I am using a classic token. Not sure about the new fine grained ones

**Comment by txbm** [comment]
> I get this error : 
> 
> MCP error -32603: Invalid arguments: content.encoding: Required, content.content: Required
> 
> I dont know if this PR can help me, does it ? 
> 
> 

Try cloning my fork locally and install it into Claude using NPX pointed at the local copy of my fork, it should work but if it doesn't just post the error and I'll fix it!

**Comment by staff0rd** [comment]
I was also getting this error, can confirm #434 fixes it. Thanks @txbm.

**Comment by cliffhall** [comment]
Deprecation Notice: Development for `server-github` has been moved to GitHub in the [http://github.com/github/github-mcp-server](https://github.com/github/github-mcp-server) repo. PRs to `server-github` in this repo are no longer being merged.
